### ISSUE TITLE
Update to Vue 3 and use composition interface

### ISSUE THREAD

**Issue by tobiasdiez** [Zettlr/Zettlr#2001]
For (better) typescript support, I would propose to upgrade to Vue 3 and use the new [Composition Interface](https://v3.vuejs.org/api/composition-api.html). (There is also a npm package to use the composition interface with vue 2 if upgrading is not possible for some reason.)

**Comment by nathanlesage** [comment]
I agree, but not now out of two reasons:

First, I'm not sure the virtual scroller already has support for Vue 3, and there are still a few more construction sites around I would like to close down before starting yet another big transition project.

Since refactoring the GUI will take considerable amount of time and planning, I'm not even sure we should do this for 2.0. So let's keep this open, but not address this before 2.0 is not at least in beta. I could really use your help around other stuff until then, if that's okay!

**Comment by tobiasdiez** [comment]
Sure, was just that I stumbled upon this and thought it would be good to record this todo somewhere.

It's probably a good idea to first upgrade Vue to version 3 (along with other dependencies) and then as a second step rewrite the components in the composition api.

**Comment by nathanlesage** [comment]
I concur, yes! Btw I just took a look at the recycle scroller, and if I saw it correctly it seems to be working with Vue 3 (since there was no apparent issue open), but it wasn't specifically released for Vue 3. So I think we should be fine. If not, we might be able to polyfill this in the process, but let's assume we don't have to!

**Comment by nathanlesage** [comment]
So, since I've now moved to Vue3 and most components already use TS, I'm going to close this issue now. The rest is still visible in the pinned issue!
